### PR TITLE
Changed jerk calculation so it is divided by seconds

### PR DESCRIPTION
**Pull request by yunfei-zheng**
Ok, so according to the [wikipedia page on Jerk](https://en.wikipedia.org/wiki/Jerk_(physics)), it says that it is the rate at which an object's acceleration changes with respect to time. So this means that you have to divide the change in acceleration by the change in time.

I know on the Kauai Labs doc page for detecting collisions it just finds the change in acceleration but I don't think that's actually right because the unit for jerk is m/s^3. If you just find the change in acceleration, that would be a unit of m/s^2, so you have to divide by seconds again to get jerk.

I think this will be more accurate than just finding the change in acceleration because jerk will take the difference in time between iterations into account. Because the difference between the iterations isn't constant if two moments have the same jerk, the change in acceleration might be different because the time between the iterations changed.

I added some comments in the file too. Also, it builds successfully. (Also we need to retest the jerk threshold for this)

### CODE REVIEW

**Comment by arnavthak** [comment]
NOPE I CONFIRMED IT WITH SOMEONE WHO TOOK CALC AND PLEASE DELETE THIS BRANCH BECAUSE THERE IS LITERALLY NO REASON TO KEEP IT

**Comment by yunfei-zheng** [comment]
## @arnavthak plz read all
Ok, so *maybe* I was wrong about the *definition* of jerk. But I still think the rate of change of acceleration would be more useful than just finding the change in acceleration. However, here is a [**literal paper in a scientific journal**](https://iopscience.iop.org/article/10.1088/0143-0807/37/6/065008/pdf) that says I'm right about the definition. Quotes:
- Page 3: " Mathematically jerk is the third derivative of our position with
respect to time "
Literally says the definition, that it is related to ***TIME***
- Page 7: "acceleration versus time onset (jerk) envelopes overlaid." 
This is saying that jerk is equivalent to acceleration over ***TIME***.
- Page 11: "corresponding jerk of about 80g*s^−1" 
This is saying jerk is measured in Gs (acceleration) over seconds (***TIME***)

*Even if* I'm not right about the definition, which I highly doubt considering it's *a literal paper in a scientific journal* that says I'm right (in addition to other sources) and just "someone who took calc" who says it's wrong, I still think finding that quantity will be more beneficial to us. 

First of all, I think this will be more accurate than just finding the change in acceleration because the rate of change will take the difference in time between iterations into account. Because the difference between the iterations isn't constant if two moments have the same jerk, the change in acceleration might be different because the time between the iterations changed.

For example, let's say you are driving the robot with a constant acceleration. Therefore, the rate of change should always be 0. However, if you just find the simple change in acceleration, the time difference between iterations will cause variance even though the acceleration hasn't changed at all, making it less accurate than the rate of change.

In conclusion, finding the rate of change of acceleration will be more accurate than just finding the change in acceleration, and is likely the actual definition of jerk. And even if you literally ignore everything I just typed, you could at least copy some of the comments into the code. Therefore, I don't think I should "DELETE THIS BRANCH BECAUSE THERE IS LITERALLY NO REASON TO KEEP IT". 

also arnav chill out with the caps bro lol

**Comment by yunfei-zheng** [comment]
## @arnavthak read this too
Here are some more sources that say I'm right:
- [Jerk: The time rate of change of acceleration](https://aapt.scitation.org/doi/abs/10.1119/1.11504)
> Literally is in the title
- [Jerk and Hyperjerk in a Rotating Frame of Reference](https://arxiv.org/pdf/1503.07051.pdf)
> First page: "Jerk is the rate of change of acceleration, that is, it is the derivative of acceleration with respect to time"
- [Mathematics of Motion Control Profiles](https://performance-motion-devices.s3.amazonaws.com/pdfs/White%20Papers/Mathematics_of_Motion_Control_Profiles.pdf)
> Second page: "Jerk is defined as the rate of change of acceleration with time."
---
Also, think about it. Jerk is the derivative of acceleration, meaning it is the rate of change of acceleration. How do you have a rate of change without dividing by time? 

Velocity is the rate of change of position. It's measured in meters per **second**. 
It tells you have many meters an object travels every second.

Acceleration is the rate of change of velocity. It's measured in meters^2 per **second**.
Every second, an object increases its velocity by *x* meters per second.

It tells you have many meters per second an object increases its velocity by every second.
Jerk is the rate of change of acceleration. Therefore, it should be measured in meters^3 per **second**. 
It tells you have many meters^2 per second an object increases its acceleration by every second.

Therefore, jerk is equal to the change in acceleration divided by the change in time.
